### PR TITLE
ADC sequence of G0 (and C0) devicec have  wrong channel shift?

### PR DESCRIPTION
**Pull request by Apehaenger**
While still working on my STM32C0x implementation #1188, I had a head-cruncher by building a ADC-Sequence example.
For me it looks like the implementation is wrong as it shifts the ADC channel only by 1 bit instead of 4.
So in my opinion ADC sequences can't work for ~~F0,~~ G0 as well as C0 devices?!

### CODE REVIEW

File: ext/modm-devices
Comment (Apehaenger) [inline ext/modm-devices:1]: Sorry, the updated submodule is only related to my C0x implementation

**Review by chris-durand — COMMENTED** [review]
Thanks for fixing this!

Register layout from the G0 manual for reference:
Screenshot_20240812_225400

The unrelated submodule update is still there. Could you remove it?

**Comment by Apehaenger** [comment]
The register layout for C0 look equal.

Regarding the unrelated submodule:
~~I would be happy if I could, but I'm unsure how ;-)
Rewind my modm-device submodule to standard and commit again?~~
Done

**Review by salkinium — APPROVED** [review]
Thanks!

**Pull request merged**